extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/MicrometerCountedInterceptor.java

```diff
@@ -0,0 +1,111 @@
+package io.quarkus.micrometer.runtime;
+
+import java.lang.reflect.Method;
+import java.util.concurrent.CompletionStage;
+
```

Comment:
Wondering if, long term, there's a better home for this kind of thing so other projects can get this OOTB?

Reply:
I don't follow. This is specific to what the @Timed / @Counted implementation for Micrometer annotations expect.. 

And I'm not sure I entirely love this. This is probably where the most changes might come as it stews.

Reply:
Agree it's specific to those annotations, but thinking from a perspective of other projects using Micrometer directly, do they need this too? If so, where should it go?

Maybe the answer is they don't, and it's fine to be only here.

Reply:
I think this commit disappeared. Happy to pull something into Micrometer core if it's generally useful.

Reply:
I think I'll keep it for now.. I just don't like it. I don't believe there is anything here that can be pulled out to somewhere more generic.